torch/csrc/profiler/python/init.cpp

```diff
@@ -122,19 +135,32 @@ void initPythonBindings(PyObject* module) {
         return py::reinterpret_borrow<py::object>(thp_device);
       });
 
-  py::class_<ExtraFields<EventType::TorchOp>>(m, "_ExtraFields_TorchOp")
-      .def_readonly("inputs", &ExtraFields<EventType::TorchOp>::inputs_)
-      .def_readonly(
-          "allow_tf32_cublas",
-          &ExtraFields<EventType::TorchOp>::allow_tf32_cublas_);
+  using torch_op_t = ExtraFields<EventType::TorchOp>;
+  py::class_<torch_op_t>(m, "_ExtraFields_TorchOp")
+      .def_readonly("inputs", &torch_op_t::inputs_)
+      .def_readonly("scope", &torch_op_t::scope_)
+      .def_readonly("sequence_number", &torch_op_t::sequence_number_)
+      .def_readonly("allow_tf32_cublas", &torch_op_t::allow_tf32_cublas_);
 
   py::class_<ExtraFields<EventType::Backend>>(m, "_ExtraFields_Backend");
 
-  py::class_<ExtraFields<EventType::Allocation>>(m, "_ExtraFields_Allocation");
+  using allocation_t = ExtraFields<EventType::Allocation>;
+  py::class_<allocation_t>(m, "_ExtraFields_Allocation")
+      .def_property_readonly(
+          "ptr",
+          [](const allocation_t& a) {
+            return reinterpret_cast<intptr_t>(a.ptr_);
+          })
+      .def_readonly("alloc_size", &allocation_t::alloc_size_)
+      .def_readonly("device_type", &allocation_t::device_type_)
+      .def_readonly("device_index", &allocation_t::device_index_)
```

Comment:
Do we want to reassemble a `device_type` and a `device_index` back into a `Device` so that this API is more accessible to end users?